common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal.logging;
+
+import org.slf4j.spi.LocationAwareLogger;
+
+import static org.slf4j.spi.LocationAwareLogger.*;
+
+/**
+ * <a href="http://www.slf4j.org/">SLF4J</a> logger which is location aware and so will log the correct origin of the
+ * logging event by filter out the wrapper itself.
+ */
+final class LocationAwareSlf4JLogger extends AbstractInternalLogger {
+
+    // IMPORTANT: All our log methods first check if the log level is enabled before call the wrapped
+    // LocationAwareLogger.log(...) method. This is done to reduce GC creation that is caused by varargs.
+
+    private static final String FQCN = LocationAwareSlf4JLogger.class.getName();
+    private static final long serialVersionUID = -8292030083201538180L;
```

Comment:
nit: could use 0 or 1 here (or your favorite number). Using the ugly id is only needed for compatibility with previous versions of the class before you added the `serialVersionUID`.

Reply:
I just pressed "generate" :D